### PR TITLE
authz: store all Perforce protect rules in `paths` column of `sub_repo_permissions` table.

### PR DESCRIPTION
**Pull request by sashaostrikov**
This is the first part of Perforce rules matching simplification. The idea is that all rules are stored as is and read one-by-one during matching (vs. previous approach of "merging" the rules before storing them in the DB).

Test plan:
Unit tests updated. Existing related tests should not fail.

Depends on and should be merged after https://github.com/sourcegraph/sourcegraph/pull/41785

Closes https://github.com/sourcegraph/sourcegraph/issues/41789

### CODE REVIEW

**Comment by pjlast** [comment]
Going to link to this comment here: https://github.com/sourcegraph/sourcegraph/pull/41785#discussion_r976464966

File: enterprise/internal/authz/perforce/protects.go [hunk 1]
```diff
@@ -423,6 +424,13 @@ func fullRepoPermsScanner(logger log.Logger, perms *authz.ExternalUserPermission
 
 				newExcludes := convertRulesForWildcardDepotMatch(match, depot, patternsToGlob)
 				srp.PathExcludes = append(srp.PathExcludes, newExcludes...)
+
+				// Adding leading "-" sign to indicate exclusion
+				formattedExcludes := make([]string, 0, len(newExcludes))
```
Comment (ryanslade) [inline enterprise/internal/authz/perforce/protects.go:429]: Do we need formattedExcludes? Can't we just append to `srp.Paths` in the for loop?

File: internal/database/sub_repo_perms_store.go [hunk 2]
```diff
@@ -71,18 +71,19 @@ func (s *subRepoPermsStore) Done(err error) error {
 // Upsert will upsert sub repo permissions data.
 func (s *subRepoPermsStore) Upsert(ctx context.Context, userID int32, repoID api.RepoID, perms authz.SubRepoPermissions) error {
```
Comment (ryanslade) [inline internal/database/sub_repo_perms_store.go:72]: Can you add tests that confirm we're writing the paths please :)

File: internal/database/sub_repo_perms_store.go [hunk 3]
```diff
@@ -91,8 +92,8 @@ SET
 // nothing is written.
 func (s *subRepoPermsStore) UpsertWithSpec(ctx context.Context, userID int32, spec api.ExternalRepoSpec, perms authz.SubRepoPermissions) error {
```
Comment (ryanslade) [inline internal/database/sub_repo_perms_store.go:93]: Can you add tests that confirm we're writing the paths please :)

File: internal/database/sub_repo_perms_store.go (hunk 2 above)
Comment (sashaostrikov) [reply internal/database/sub_repo_perms_store.go:72]: good point!

File: internal/database/sub_repo_perms_store.go (hunk 3 above)
Comment (sashaostrikov) [reply internal/database/sub_repo_perms_store.go:93]: good point!

File: enterprise/internal/authz/perforce/protects.go (hunk 1 above)
Comment (sashaostrikov) [reply enterprise/internal/authz/perforce/protects.go]: lol, true 🤦🏻

**Review by pjlast — APPROVED** [review]
LGTM 👍

**Pull request merged**